tests/integration/test_cloudformation.py

```diff
@@ -1961,40 +1961,46 @@ def test_deploy_stack_with_sub_select_and_sub_getaz(self):
         metric_alarms = cw_client.describe_alarms().get("MetricAlarms", [])
         composite_alarms = cw_client.describe_alarms().get("CompositeAlarms", [])
 
-        # deploy stack
-        create_and_await_stack(StackName=stack_name, TemplateBody=template)
-        exports = cfn_client.list_exports()["Exports"]
+        def _assert_resources_created():
+            exports = cfn_client.list_exports()["Exports"]
+            subnets = [export for export in exports if export["Name"] == "public-sn-a"]
+            instances = [export for export in exports if export["Name"] == "RegmonEc2InstanceId"]
 
-        subnets = [export for export in exports if export["Name"] == "public-sn-a"]
-        instances = [export for export in exports if export["Name"] == "RegmonEc2InstanceId"]
+            self.assertEqual(1, len(subnets))
+            self.assertEqual(1, len(instances))
 
```

Comment:
QA nit:

Checking the exports without using randomized export names potentially leads to conflicts in case of parallel test execution. Currently not a problem but might be a blocker when/if we ever move to more parallelized testing.
Using stack outputs would be fine since they're scoped to the stack.

Reply:
Great points 👍  - will add a `TODO` comment before merging.